File: app/src/main/java/com/ebsoft/watchlist/di/ActivityBuilder.java
package com.ebsoft.watchlist.di;

import com.ebsoft.watchlist.ui.main.MainActivity;

import dagger.Module;
import dagger.android.ContributesAndroidInjector;

/**
 * Created by endre on 07/09/18.
 */

@Module
public abstract class ActivityBuilder {

    @ContributesAndroidInjector
    abstract MainActivity bindMainActivity();
}

File: app/src/main/java/com/ebsoft/watchlist/di/ActivityBuilder.kt
package com.ebsoft.watchlist.di

import com.ebsoft.watchlist.ui.main.MainActivity

import dagger.Module
import dagger.android.ContributesAndroidInjector

/**
 * Created by endre on 07/09/18.
 */

@Module
abstract class ActivityBuilder {

    @ContributesAndroidInjector
    internal abstract fun bindMainActivity(): MainActivity
}
